Server/ReasnAPI/ReasnAPI/Services/ImageService.cs

```diff
@@ -1,16 +1,175 @@
+    {
+        if (!imageDtos.Any())
+        {
+            throw new ArgumentException("No images provided");
+        }
+
+        var objectType = imageDtos[0].ObjectType;
+
+        if (objectType == ObjectType.User)
+        {
+            if (imageDtos.Count != 1)
+            {
+                throw new ArgumentException("For User type, only one image can be updated");
+            }
+
+            var image = context.Images.FirstOrDefault(r => r.ObjectId == objectId && r.ObjectType == ObjectType.User);
+            if (image is null)
+            {
+                throw new NotFoundException("Image not found");
+            }
+
+            image.ImageData = imageDtos[0].ImageData;
+
+            context.Images.Update(image);
+        }
+        else if (objectType == ObjectType.Event)
+        {
+            var images = context.Images.Where(r => r.ObjectId == objectId && r.ObjectType == ObjectType.Event).ToList();
+
+            // Remove images that are not in the new list
+            foreach (var image in images)
+            {
+                if (!imageDtos.Any(dto => dto.ImageData == image.ImageData))
+                {
+                    context.Images.Remove(image);
+                }
+            }
+
+            // Add new images that are not in the database
+            var newImages = new List<Image>();
+            foreach (var imageDto in imageDtos)
+            {
+                if (!images.Any(img => img.ImageData == imageDto.ImageData))
+                {
+                    var newImage = imageDto.ToEntity();
+                    newImages.Add(newImage);
+                }
+            }
+            context.Images.AddRange(newImages);
+        }
+
+        context.SaveChanges();
+    }
+
+    public void DeleteImage(int id)
+    {
+        var image = context.Images.FirstOrDefault(r => r.Id == id);
+        if (image is null)
+        {
+            throw new NotFoundException("Image not found");
+        }
+
+        context.Images.Remove(image);
+        context.SaveChanges();
+    }
 
-namespace ReasnAPI.Services {
-    public class ImageService (ReasnContext context) {
-        private readonly ReasnContext _context = context;
+    public ImageDto GetImageById(int id)
+    {
+        var image = context.Images.Find(id);
+        if (image is null)
+        {
+            throw new NotFoundException("Image not found");
+        }
+
+        var imageDto = image.ToDto();
+        
+        return imageDto;
+    }
+
+    public IEnumerable<ImageDto> GetImagesByObjectId(int objectId, ObjectType objectType)
+    {
+        var images = context.Images
+            .Where(image => image.ObjectId == objectId && image.ObjectType == objectType)
+            .ToList();
+
+        if (!images.Any())
+        {
+            throw new NotFoundException("Images not found");
+        }
+
+        var imageDtos = images.Select(image => image.ToDto()).AsEnumerable();

```

Comment:
.toDtoList wouldn't work?

Reply:
changed, i should have missed that earlier